app/routes.py

```diff
@@ -65,15 +65,19 @@ def submit():
     if submitForm.validate_on_submit():
         createNewSubmission(current_user.id, challenge["id"], pathString, int(time.time()))
         return redirect(url_for('main.view', id=int(submitForm.challenge_id.data)))
-    return render_template('view.html', form=form, submitForm=submitForm, submitted=False, challenge=challenge, errors=submitForm.errors["path"][0], path=pathString)
+    return redirect(url_for("main.view", id=int(submitForm.challenge_id.data), errors=submitForm.errors["path"][0], path=pathString))
```

Comment:
Not part of PR scope but now that id is part of website link, a check to see if the id is valid would be good.

Reply:
challenge ID validation occurs at the start of view

Reply:
but yes i see what you mean

Reply:
It checks for if it's -1, not if it's some number like 999999, and having ?id=999999 will throw a 500 error rather than redirecting it to main

Reply:
would getChallenge(id) is not none work?

Reply:
Yes

Reply:
Change pushed!